app/src/features/apiClient/contexts/apiClient.tsx

```diff
@@ -162,20 +174,34 @@ export const ApiClientProvider: React.FC<ApiClientProviderProps> = ({ children }
 
   const onNewRecord = useCallback((apiClientRecord: RQAPI.Record) => {
     setApiClientRecords((prev) => {
-      return [...prev, { ...apiClientRecord }];
+      const updatedRecords = [...prev, { ...apiClientRecord }];
+
+      // Tracking user properties for analytics
+      trackUserProperties(updatedRecords);
+      return updatedRecords;
```

Comment:
Instead of adding tracking call to each state update can we listen to state updates using `useEffect` and whenever `apiClientRecords` change, we can log the properties. WDYT?